pkg/cmd/roachtest/spec/option.go

```diff
@@ -38,6 +38,19 @@ func CPU(n int) Option {
 	}
 }
 
+// WorkloadNode indicates that the last node is a workload node.
+func WorkloadNode() Option {
```

Comment:
instead of having this value as boolean, can we make with number of nodes as input. This way, we need not use the last node from cluster. Use this number as the number of nodes needed for workload. This gives us the flexibility to use any number of nodes as workload for future.

Reply:
> can we make with number of nodes as input. This way, we need not use the last node from cluster

This was actually my original approach since it seems like the "cleaner" solution but I scrapped it after partially implementing it. All tests that currently use a workload already provision an extra node and assign it as the workload node. I didn't want to change people's mental model. There's also a lot of node arithmetic that's done in roachtests around the workload node, and it seemed like a lot of work to go through all tests and change it. Vs. the current impl should be a drop in replacement.

That being said, allowing more than one workload node seems orthogonal from separating workload nodes from crdb nodes and could be useful. Could just make it so the last n nodes are workload nodes. WDYT? Although i'm skeptical if I should add it until we have a test that wants to use more than one workload node.

Reply:
Sure! Makes sense. Thanks!

Reply:
IMO, the main use case for multiple workload nodes would be for multi-region tests where we want to simulate a more realistic deployment of multi-region workloads connected to their regional clusters. But this can be done in future work.